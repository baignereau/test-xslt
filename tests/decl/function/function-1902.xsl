<?xml version="1.0"?> 

<?spec xslt#stylesheet-functions?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0"
  xmlns:fos="http://www.w3.org/xpath-functions/spec/namespace"
  xmlns:math="http://www.w3.org/2005/xpath-functions/math" exclude-result-prefixes="fos math">

  <!-- test that all new functions listed in the F+O specification are implemented (or claim to be available!) in XSLT 2.0 -->
  <!-- This test uses function-catalog.xml as input. This should be updated from time to time
     with the latest version -->

  <xsl:template match="/">
    <out>
      <xsl:apply-templates
        select=".//fos:proto[not(../../fos:opermap)][not(../../@prefix=('math', 'map'))]"/>
    </out>
  </xsl:template>


  <xsl:template match="fos:proto">
    <xsl:variable name="lexname"
      select="if (../../@prefix) then concat(../../@prefix, ':', @name) else @name"/>
    <f name="{$lexname}" arity="{count(fos:arg)}"
      test="{function-available($lexname, count(fos:arg))}"/>
  </xsl:template>
</xsl:stylesheet>
