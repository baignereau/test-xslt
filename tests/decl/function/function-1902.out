<?xml version="1.0" encoding="UTF-8"?><out><f name="node-name" arity="0" test="false"/><f name="node-name" arity="1" test="true"/><f name="nilled" arity="1" test="true"/><f name="string" arity="0" test="true"/><f name="string" arity="1" test="true"/><f name="data" arity="0" test="false"/><f name="data" arity="1" test="true"/><f name="base-uri" arity="0" test="true"/><f name="base-uri" arity="1" test="true"/><f name="document-uri" arity="0" test="false"/><f name="document-uri" arity="1" test="true"/><f name="error" arity="0" test="true"/><f name="error" arity="1" test="true"/><f name="error" arity="2" test="true"/><f name="error" arity="3" test="true"/><f name="trace" arity="2" test="true"/><f name="abs" arity="1" test="true"/><f name="ceiling" arity="1" test="true"/><f name="floor" arity="1" test="true"/><f name="round" arity="1" test="true"/><f name="round" arity="2" test="false"/><f name="round-half-to-even" arity="1" test="true"/><f name="round-half-to-even" arity="2" test="true"/><f name="format-integer" arity="2" test="false"/><f name="format-integer" arity="3" test="false"/><f name="format-number" arity="2" test="true"/><f name="format-number" arity="3" test="true"/><f name="codepoints-to-string" arity="1" test="true"/><f name="string-to-codepoints" arity="1" test="true"/><f name="compare" arity="2" test="true"/><f name="compare" arity="3" test="true"/><f name="codepoint-equal" arity="2" test="true"/><f name="concat" arity="3" test="true"/><f name="string-join" arity="1" test="false"/><f name="string-join" arity="2" test="true"/><f name="substring" arity="2" test="true"/><f name="substring" arity="3" test="true"/><f name="string-length" arity="0" test="true"/><f name="string-length" arity="1" test="true"/><f name="normalize-space" arity="0" test="true"/><f name="normalize-space" arity="1" test="true"/><f name="normalize-unicode" arity="1" test="true"/><f name="normalize-unicode" arity="2" test="true"/><f name="upper-case" arity="1" test="true"/><f name="lower-case" arity="1" test="true"/><f name="translate" arity="3" test="true"/><f name="encode-for-uri" arity="1" test="true"/><f name="iri-to-uri" arity="1" test="true"/><f name="escape-html-uri" arity="1" test="true"/><f name="contains" arity="2" test="true"/><f name="contains" arity="3" test="true"/><f name="starts-with" arity="2" test="true"/><f name="starts-with" arity="3" test="true"/><f name="ends-with" arity="2" test="true"/><f name="ends-with" arity="3" test="true"/><f name="substring-before" arity="2" test="true"/><f name="substring-before" arity="3" test="true"/><f name="substring-after" arity="2" test="true"/><f name="substring-after" arity="3" test="true"/><f name="matches" arity="2" test="true"/><f name="matches" arity="3" test="true"/><f name="replace" arity="3" test="true"/><f name="replace" arity="4" test="true"/><f name="tokenize" arity="2" test="true"/><f name="tokenize" arity="3" test="true"/><f name="analyze-string" arity="2" test="false"/><f name="analyze-string" arity="3" test="false"/><f name="resolve-uri" arity="1" test="true"/><f name="resolve-uri" arity="2" test="true"/><f name="true" arity="0" test="true"/><f name="false" arity="0" test="true"/><f name="boolean" arity="1" test="true"/><f name="not" arity="1" test="true"/><f name="years-from-duration" arity="1" test="true"/><f name="months-from-duration" arity="1" test="true"/><f name="days-from-duration" arity="1" test="true"/><f name="hours-from-duration" arity="1" test="true"/><f name="minutes-from-duration" arity="1" test="true"/><f name="seconds-from-duration" arity="1" test="true"/><f name="dateTime" arity="2" test="true"/><f name="year-from-dateTime" arity="1" test="true"/><f name="month-from-dateTime" arity="1" test="true"/><f name="day-from-dateTime" arity="1" test="true"/><f name="hours-from-dateTime" arity="1" test="true"/><f name="minutes-from-dateTime" arity="1" test="true"/><f name="seconds-from-dateTime" arity="1" test="true"/><f name="timezone-from-dateTime" arity="1" test="true"/><f name="year-from-date" arity="1" test="true"/><f name="month-from-date" arity="1" test="true"/><f name="day-from-date" arity="1" test="true"/><f name="timezone-from-date" arity="1" test="true"/><f name="hours-from-time" arity="1" test="true"/><f name="minutes-from-time" arity="1" test="true"/><f name="seconds-from-time" arity="1" test="true"/><f name="timezone-from-time" arity="1" test="true"/><f name="adjust-dateTime-to-timezone" arity="1" test="true"/><f name="adjust-dateTime-to-timezone" arity="2" test="true"/><f name="adjust-date-to-timezone" arity="1" test="true"/><f name="adjust-date-to-timezone" arity="2" test="true"/><f name="adjust-time-to-timezone" arity="1" test="true"/><f name="adjust-time-to-timezone" arity="2" test="true"/><f name="format-dateTime" arity="5" test="true"/><f name="format-dateTime" arity="2" test="true"/><f name="format-date" arity="5" test="true"/><f name="format-date" arity="2" test="true"/><f name="format-time" arity="5" test="true"/><f name="format-time" arity="2" test="true"/><f name="resolve-QName" arity="2" test="true"/><f name="QName" arity="2" test="true"/><f name="prefix-from-QName" arity="1" test="true"/><f name="local-name-from-QName" arity="1" test="true"/><f name="namespace-uri-from-QName" arity="1" test="true"/><f name="namespace-uri-for-prefix" arity="2" test="true"/><f name="in-scope-prefixes" arity="1" test="true"/><f name="name" arity="0" test="true"/><f name="name" arity="1" test="true"/><f name="local-name" arity="0" test="true"/><f name="local-name" arity="1" test="true"/><f name="namespace-uri" arity="0" test="true"/><f name="namespace-uri" arity="1" test="true"/><f name="number" arity="0" test="true"/><f name="number" arity="1" test="true"/><f name="lang" arity="1" test="true"/><f name="lang" arity="2" test="true"/><f name="root" arity="0" test="true"/><f name="root" arity="1" test="true"/><f name="index-of" arity="2" test="true"/><f name="index-of" arity="3" test="true"/><f name="empty" arity="1" test="true"/><f name="exists" arity="1" test="true"/><f name="distinct-values" arity="1" test="true"/><f name="distinct-values" arity="2" test="true"/><f name="insert-before" arity="3" test="true"/><f name="remove" arity="2" test="true"/><f name="head" arity="1" test="false"/><f name="tail" arity="1" test="false"/><f name="reverse" arity="1" test="true"/><f name="subsequence" arity="2" test="true"/><f name="subsequence" arity="3" test="true"/><f name="unordered" arity="1" test="true"/><f name="zero-or-one" arity="1" test="true"/><f name="one-or-more" arity="1" test="true"/><f name="exactly-one" arity="1" test="true"/><f name="deep-equal" arity="2" test="true"/><f name="deep-equal" arity="3" test="true"/><f name="count" arity="1" test="true"/><f name="avg" arity="1" test="true"/><f name="max" arity="1" test="true"/><f name="max" arity="2" test="true"/><f name="min" arity="1" test="true"/><f name="min" arity="2" test="true"/><f name="sum" arity="1" test="true"/><f name="sum" arity="2" test="true"/><f name="id" arity="1" test="true"/><f name="id" arity="2" test="true"/><f name="element-with-id" arity="1" test="true"/><f name="element-with-id" arity="2" test="true"/><f name="idref" arity="1" test="true"/><f name="idref" arity="2" test="true"/><f name="doc" arity="1" test="true"/><f name="doc-available" arity="1" test="true"/><f name="collection" arity="0" test="true"/><f name="collection" arity="1" test="true"/><f name="uri-collection" arity="0" test="false"/><f name="uri-collection" arity="1" test="false"/><f name="unparsed-text" arity="1" test="true"/><f name="unparsed-text" arity="2" test="true"/><f name="unparsed-text-available" arity="1" test="true"/><f name="unparsed-text-available" arity="2" test="true"/><f name="environment-variable" arity="1" test="false"/><f name="available-environment-variables" arity="0" test="false"/><f name="generate-id" arity="0" test="true"/><f name="generate-id" arity="1" test="true"/><f name="parse-xml" arity="1" test="false"/><f name="parse-xml" arity="2" test="false"/><f name="serialize" arity="1" test="false"/><f name="serialize" arity="2" test="false"/><f name="position" arity="0" test="true"/><f name="last" arity="0" test="true"/><f name="current-dateTime" arity="0" test="true"/><f name="current-date" arity="0" test="true"/><f name="current-time" arity="0" test="true"/><f name="implicit-timezone" arity="0" test="true"/><f name="default-collation" arity="0" test="true"/><f name="static-base-uri" arity="0" test="true"/><f name="function-name" arity="1" test="false"/><f name="function-arity" arity="1" test="false"/><f name="map" arity="2" test="false"/><f name="filter" arity="2" test="false"/><f name="fold-left" arity="3" test="false"/><f name="fold-right" arity="3" test="false"/><f name="map-pairs" arity="3" test="false"/></out>